Added Video Functionality

PrabhutvaAgrawal opened the pull request:
I have tried to add Video Functionality to the app by creating a new fragment. It will be able to play YouTube and FaceBook videos when video url is passed. Currently connenction to firebase is left and code is still incomplete.

PrabhutvaAgrawal committed:
Added Video Functionality

aryanagarwal15 committed:
fixed bottom nav

Rushi98 commented on androidClient/app/src/main/java/com/csatimes/dojma/activities/VideosActivity.java:
```diff
@@ -0,0 +1,60 @@
+        setContentView(R.layout.activity_videos);
+        //getActionBar().hide();
+        videoID= getIntent().getStringExtra("id");
+        String dateStamp = getIntent().getStringExtra("date");
```
create a `public static` method in **`VideosActivity`** class to launch the activity with required parameters. (date, title, etc)

Rushi98 commented on androidClient/app/src/main/java/com/csatimes/dojma/activities/VideosActivity.java:
```diff
@@ -0,0 +1,60 @@
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_videos);
+        //getActionBar().hide();
```
delete code that is not needed. Don't comment it out.

Rushi98 commented on androidClient/app/src/main/java/com/csatimes/dojma/adapters/VideosAdapter.java:
```diff
@@ -0,0 +1,44 @@
+        return holder;
+    }//link xml to recycler view
+
+    @Override//means whatever we are extending is changed to put our own stuff
```
lol, wkt

Rushi98 commented on androidClient/app/src/main/java/com/csatimes/dojma/fragments/VideosFragment.java:
```diff
@@ -0,0 +1,78 @@
+    //private static final String TAG = TAG_PREFIX + VideosFragment.class.getSimpleName();
+    private RecyclerView videosRv;
+
+    String videoURL[] = {"https://www.facebook.com/DoPy.BITSGoa/videos/622644644796899/","https://www.facebook.com/bitsquark/videos/309257229884592/","https://www.facebook.com/bitsquark/videos/2300637253556600/","https://www.youtube.com/watch?v=IM2czqavlWM", "https://www.youtube.com/watch?v=737dCFBqO4I", "https://www.youtube.com/watch?v=yuboWIBPDvk", "https://www.youtube.com/watch?v=pXCpc2Bcirs", ...
```
create a new class (YoutubeVideo, maybe) to hold these data members (url, title, etc)

Rushi98 commented on androidClient/app/src/main/java/com/csatimes/dojma/fragments/VideosFragment.java:
```diff
@@ -0,0 +1,78 @@
+    public void onStart() {
+        super.onStart();
+        final Activity activity = getActivity();
+            final Context context = getContext();
```
use consistent alignment

PrabhutvaAgrawal committed:
Resolved some bugs

Fixed activity not finishing after closing.
Fixed activity reloading after rotating screen.
Added Share button and Back button.
Minor changes to interface.

aryanagarwal15 committed:
added firebase

aryanagarwal15 commented:
@tanmaydixit @Rushi98 Please merge this PR. I've completed everything required for the feature

aryanagarwal15 commented:
@tanmaydixit please merge!

Rushi98 committed:
maintenance

- bumped library versions
- cleaned up some bits of code
